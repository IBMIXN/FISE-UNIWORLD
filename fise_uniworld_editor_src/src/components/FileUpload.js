import PropTypes from "prop-types";
import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import PublishIcon from "@material-ui/icons/Publish";

const useStyles = makeStyles((theme) => ({
  fileUpload: {
    width: "0.1px",
    height: "0.1px",
    opacity: 0,
    overflow: "hidden",
    position: "absolute",
    zIndex: -1,
  },
  fileUploadLabel: {
    "&:focus": {
      outline: "1px dotted #000",
    },
    fontSize: "1rem",
    fontWeight: 700,
    color: "#fff",
    backgroundColor: theme.palette.info.main,
    display: "inline-block",
    padding: theme.spacing(1),
    cursor: "pointer",
    width: "180px",
    paddingTop: "14px",
  },
}));

const FileUpload = ({ name, onChange, width, children, accept }) => {
  const classes = useStyles();
  return (
    <React.Fragment>
      <input
        type="file"
        id={name}
        name={name}
        className={classes.fileUpload}
        onChange={onChange}
        accept={accept}
      />
      <label htmlFor={name} className={classes.fileUploadLabel} style={width && { width: width }}>
        <PublishIcon fontSize="small" style={{ paddingTop: "4px" }} />
        {children}
      </label>
    </React.Fragment>
  );
};

FileUpload.propTypes = {
  children: PropTypes.node,
  name: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  width: PropTypes.string,
  accept: PropTypes.string,
};

export default FileUpload;
