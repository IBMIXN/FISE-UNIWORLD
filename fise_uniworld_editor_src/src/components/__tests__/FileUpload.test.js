import React from 'react';
import { shallow } from 'enzyme';
import FileUpload from '../FileUpload';

describe('FileUpload test', () => {
  it('FileUpload should match snapshot', () => {
    const component = shallow(
      <FileUpload
        name='mockName'
        onChange={() => {}}
        width='100px'
        accept='mockAccept'
      />
    );
    expect(component).toMatchSnapshot();
  });
});
