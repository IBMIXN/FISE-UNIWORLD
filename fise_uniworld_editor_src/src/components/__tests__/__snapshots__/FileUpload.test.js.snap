// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileUpload test FileUpload should match snapshot 1`] = `
<Fragment>
  <input
    accept="mockAccept"
    className="makeStyles-fileUpload-1"
    id="mockName"
    name="mockName"
    onChange={[Function]}
    type="file"
  />
  <label
    className="makeStyles-fileUploadLabel-2"
    for="mockName"
    style={
      Object {
        "width": "100px",
      }
    }
  >
    <Memo([object Object])
      fontSize="small"
      style={
        Object {
          "paddingTop": "4px",
        }
      }
    />
  </label>
</Fragment>
`;
